roipoly/roipoly.py

```diff
@@ -69,15 +69,18 @@ def __init__(self, fig=None, ax=None, color='b',
         self.__cid2 = self.fig.canvas.mpl_connect(
             'button_press_event', self.__button_press_callback)
 
+        # determine the type of environment
+        using_spyder = any(['SPYDER' in x for x in os.environ])
+        self.interactive = sys.flags.interactive or using_spyder
+
         if show_fig:
             self.show_figure()
 
-    @staticmethod
-    def show_figure():
-        if sys.flags.interactive:
+    def show_figure(self):
+        while (not self.completed) and self.interactive:
             plt.show(block=False)
-        else:
-            plt.show(block=True)
+            plt.pause(0.1)
```

Comment:
Hm, if I understand correctly, this would issue a plt.show() every 0.1 sec. What exactly are you trying to do here? Can this be solved more efficiently?

Reply:
The only way I could get this to work on spyder was via this method. I think it is to do with two layers of qt blocking (matplotlib and spyder). There is more on this issue (and where I found the fix) [here](https://github.com/spyder-ide/spyder/issues/4367).

Reply:
Interesting. It feels a bit hacky though. I will have a closer look at this soon